### PR TITLE
Update DG - Use Case and User Stories

### LINKED ISSUE #496: Update JY's DG Use Cases and Stories
Check through with actual behaviour of the app and update
DG use case and user story sections accordingly. 
Also compare behaviour with manual testing section of DG.

### PR DESCRIPTION
**Pull request by jianyangg**
Slight edits to each.

Close #496

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1027,31 +1028,33 @@ otherwise)
...
+- Password will be changed.
 
 **MSS:**
 
 1. Registered owner opens the HomeBoss application.
-2. Registered owner enters the account recovery command without any command flags (i.e., `--answer`).
-3. US displays the forget password question that the user set during account registration.
+2. Registered owner enters the account recovery command without any command flags (i.e., no `--answer`).
+3. US displays the "forget password" secret question that the user set during account registration, together with the
+   command format for account recovery.
```
Comment (B-enguin) [inline docs/DeveloperGuide.md:1117]: Would this be considered giving too much details about the mechanics of the system, i.e. "--answer"

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -1027,31 +1028,33 @@ otherwise)
 
 **Actor:** Registered owner
 
-**Preconditions:** Registered owner is logged out.
+**Preconditions:** An owner is registered with HomeBoss.
 
 **Guarantees:**
 
-- Password would be changed.
+- Password will be changed.
```
Comment (B-enguin) [inline docs/DeveloperGuide.md:1110]: Would it be better to specify if the command is executed successfully

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -1354,15 +1357,21 @@ otherwise)
 
 **Extensions:**
 
-- 1a. Logged-in Owner specifies invalid customer.
+- 1a. Logged-in Owner specifies a customer ID that does not exist is a positive integer.
```
Comment (B-enguin) [inline docs/DeveloperGuide.md:1434]: Would this be considered too much internal details. Also I think the phrasing is a bit weird.

**Review by B-enguin — COMMENTED** [review]
Some general comments

**Review by Gabriel4357 — COMMENTED** [review]
LGTM

**Review by Gabriel4357 — APPROVED** [review]
LGTM

**Pull request merged**